tasks/update-cnb-dependency/generate_commit_message.go

```diff
@@ -7,40 +7,61 @@ import (
 )
 
 func GenerateCommitMessage(oldDeps, newDeps Dependencies, dep Dependency, storyID int) string {
-	added := false
-	rebuilt := false
+	added, rebuilt, stacks := findAddedDeps(dep, oldDeps, newDeps)
+
+	removedDeps, stacks := findRemovedDeps(dep, oldDeps, newDeps, stacks)
+
+	return formCommitMessage(dep, added, rebuilt, removedDeps, stacks, storyID)
+}
+
+func findAddedDeps(dep Dependency, oldDeps, newDeps Dependencies) (bool, bool, map[string]bool) {
+	var added, rebuilt bool
+	// Use maps to ensure no duplicates
```

Comment:
Could we choose a better name for this variable to remove the need for a comment? Perhaps something like `uniqueStacks`?

Reply:
Changed